types/nightwatch/index.d.ts

```diff
@@ -6418,7 +6569,15 @@ export interface WebDriverProtocolCookies {
      * @see deleteCookie
      * @see deleteCookies
      */
-    cookie(method: string, callbackOrCookie?: () => void): this;
+    cookie(
+        method: "GET" | "POST" | "DELETE",
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<Cookie[] | null>) => void
+    ): Awaitable<this, Cookie[] | null>;
+    cookie(
+        method: "GET" | "POST" | "DELETE",
+        cookie: Cookie,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<Cookie[] | null>) => void
+    ): Awaitable<this, Cookie[] | null>;
```

Comment:
This should have three overloads:
```suggestion
    cookie(
        method: "GET" | "DELETE",
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<Cookie[] | null>) => void
    ): Awaitable<this, Cookie[] | null>;
    cookie(
        method: "POST",
        cookie: Cookie,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void
    ): Awaitable<this, null>;
    cookie(
        method: "DELETE",
        cookieName: string,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void
    ): Awaitable<this, null>;
```